tests/test_optimizer.py

```diff
@@ -342,6 +342,18 @@ def test_qualify_columns__invalid(self):
                     )
                     optimizer.qualify_columns.validate_qualify_columns(expression)
 
+        # Cannot expand star as not all nested structs are aliased
+        self.assertEqual(
```

Comment:
why is this here and not in the qualify_columns test?

Reply:
Not being able to expand the star doesn't mean it's an invalid query. I'd just add this in `qualify_columns.sql`, which already contains similar tests:

```
# execute: false
SELECT SOME_UDF(data).* FROM t;
SELECT SOME_UDF(t.data).* FROM t AS t;
```